staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go

```diff
@@ -29,11 +29,19 @@ import (
 func WithPanicRecovery(handler http.Handler) http.Handler {
 	return withPanicRecovery(handler, func(w http.ResponseWriter, req *http.Request, err interface{}) {
 		if err == http.ErrAbortHandler {
-			// honor the http.ErrAbortHandler sentinel panic value:
-			//   ErrAbortHandler is a sentinel panic value to abort a handler.
-			//   While any panic from ServeHTTP aborts the response to the client,
-			//   panicking with ErrAbortHandler also suppresses logging of a stack trace to the server's error log.
-			return
+			// Honor the http.ErrAbortHandler sentinel panic value
+			//
+			// If ServeHTTP panics, the server (the caller of ServeHTTP) assumes
+			// that the effect of the panic was isolated to the active request.
+			// It recovers the panic, logs a stack trace to the server error log,
+			// and either closes the network connection or sends an HTTP/2
+			// RST_STREAM, depending on the HTTP protocol. To abort a handler so
+			// the client sees an interrupted response but the server doesn't log
+			// an error, panic with the value ErrAbortHandler.
+			//
+			// We are throwing http.ErrAbortHandler deliberately so that a client is notified
+			klog.Warningf("Aborting the request since ErrAbortHandler was received (%v) on %v %v. This might result in a EOF error on the client side", err, req.Method, req.RequestURI)
```

Comment:
I'd prefer a metric, or a rate limit on this log message like that in https://github.com/kubernetes/kubernetes/pull/88600/files

Reply:
+1 for metric.

What do we have now for errors like EOF or tls errors with `--v=2`? They are also visible, aren't they? A warning here is definitely too heavy. We won't warn about EOFs either, will we?

Reply:
What will we see in `--v=2` for timeouts in general?